src/Microsoft.AspNet.Mvc.Core/ActionResults/ContentResult.cs

```diff
@@ -21,7 +22,9 @@ public class ContentResult : ActionResult
 
             if (!string.IsNullOrEmpty(ContentType))
             {
-                response.ContentType = ContentType;
+                var contentTypeHeader = new MediaTypeHeaderValue(ContentType);
+                contentTypeHeader.Encoding = ContentEncoding;
+                response.ContentType = contentTypeHeader.ToString();
             }
 
             if (Content != null)
```

Comment:
You're in an `async Task` method, the compiler will end the task implicitly at the end of the method. The server will send the headers at the end of the task chain if they haven't been sent yet.